lib/application_insights/rack/track_request.rb

```diff
@@ -82,6 +86,24 @@ def format_request_duration(duration_seconds)
 
         Time.at(duration_seconds).gmtime.strftime("00.%H:%M:%S.%7N")
       end
+
+      def request_id_header(env)
+        id = SecureRandom.hex(4)
+        request_id_header = env['HTTP_REQUEST_ID']
+
+        if request_id_header
+          request_id_starts_correctly = request_id_header[0] == '|'
+
+          if request_id_starts_correctly
+            request_id_has_end = %w(. _).include?(request_id_header[-1])
+            request_id_header << '.' unless request_id_has_end
+
+            return "#{request_id_header}#{id}."
+          end
+        end
+
+        "|#{id}."
```

Comment:
perhaps when you generating new request ID, not appending to existing - you need to generate it separately and make longer.

Reply:
I've updated my implementation - it will generate 16 chars unless it is appending, in which case, it will generate 8